standalone_scripts/ttalps_plotter.py

```diff
@@ -12,85 +13,79 @@ def setupLegend(legend):
   legend.SetTextFont(42)
   legend.SetTextSize(0.035)
 
-def getStackandLegendDicts(config,file_type,efficiency=False):
-  hists_dict = {}
-  legends_dict = {}
-  legend_position = config.signal_legend_position
-  if(file_type=="background"):
-    legend_position = config.background_legend_position
-
-  if(efficiency == False):
-    for variable, params in config.variables.items():
-      hists_dict[variable] = ROOT.THStack(variable+file_type,variable+file_type)
-
-      legend = TLegend(legend_position[0],legend_position[1],legend_position[2],legend_position[3])
-      setupLegend(legend)
-      legends_dict[variable] = legend
-  else:
-    for efficiency, eff_params in config.efficiency_plots.items():
-      variable, title, xlabel, ylabel = eff_params
-      hists_dict[efficiency] = ROOT.THStack(variable+file_type+"_eff",variable+file_type+"_eff")
-
-      legend = TLegend(legend_position[0],legend_position[1],legend_position[2],legend_position[3])
-      setupLegend(legend)
-      legends_dict[efficiency] = legend
 
-  return hists_dict, legends_dict
+def getStackDict(config, file_type, efficiency=False):
+  hists_dict = {}
+  variables = config.efficiency_plots if efficiency else config.variables
 
+  for name, params in variables.items():
+    variable = params[0] if efficiency else name
+    title = variable + file_type
+    if efficiency:
+      title += "_eff"
+    hists_dict[name] = ROOT.THStack(title, title)
 
-def getTotalEntires(hist):
-  n_events = 0
-  for bin in range(1,hist.GetNbinsX()+1):
-    n_events += hist.GetBinContent(bin)
-  return n_events
+  return hists_dict
 
 
-def getNBackgroundEvents(config):
-  tot_background_events = {}
-  for name, fileInfo in config.files.items():
-    filename, file_type = fileInfo
-    if(file_type!="background"):
-      continue
-    input_path = config.background_path
-    input_file = TFile.Open(input_path+"/"+filename, "READ")
-    for variable, params in config.variables.items():
-      hist = input_file.Get(variable)
-      n_events = getTotalEntires(hist)
-      tot_background_events[variable] = n_events
+def getLegendDicts(config, file_type, efficiency=False):
+  legends_dict = {}
+  legend_position = config.legend_position[file_type]
+  variables = config.efficiency_plots if efficiency else config.variables
+  
+  for name in variables.keys():
+    legend = TLegend(legend_position[0], legend_position[1], legend_position[2], legend_position[3])
+    setupLegend(legend)
+    legends_dict[name] = legend
 
-  return tot_background_events
+  return legends_dict
 
 
-def setupHist(hist,config,params,linestyles,file_type,norm_scale=-1):
-  title,logy,norm1,rebin,xmin,xmax,ymin,ymax,xlabel,ylabel = params
+def setupHist(hist, params, linestyles, file_type, norm_scale=-1):
   line_color, line_style = linestyles
-  if(norm1):
+  
+  norm1 = params[2]
+  if (norm1):
```

Comment:
I think the different ways to normalize has to be updated here. When I started plotting I only normalized to 1 and norm1 (sorry for the unclear naming.. ) = normalize to 1. Then I had the option to still use this norm_scale because we need that for the background to make the proportions of background correct according to cross sections when stacking.